src/OpenTelemetry.Instrumentation.EventCounters/EventCounterListener.cs

```diff
@@ -0,0 +1,229 @@
+        }
+
+        private Dictionary<string, string> EnableEventArgs => new Dictionary<string, string> { ["EventCounterIntervalSec"] = this.options.RefreshIntervalSecs.ToString(), };
+
+        protected override void OnEventSourceCreated(EventSource source)
+        {
+            // TODO: Add Configuration options to selectively subscribe to EventCounters
+            try
+            {
+                if (!this.isInitialized)
+                {
+                    this.eventSources.Add(source);
+                }
+                else
+                {
+                    this.EnableEvents(source, EventLevel.Verbose, EventKeywords.All, this.EnableEventArgs);
+                }
+            }
+            catch (Exception ex)
+            {
+                EventCountersInstrumentationEventSource.Log.ErrorEventCounter(source.Name, ex.Message);
+            }
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            if (!this.isInitialized || !eventData.EventName.Equals("EventCounters"))
+            {
+                return;
+            }
+
+            try
+            {
+                if (eventData.Payload.Count > 0 && eventData.Payload[0] is IDictionary<string, object> eventPayload)
+                {
+                    this.ExtractAndPostMetric(eventData.EventSource.Name, eventPayload);
+                }
+                else
+                {
+                    EventCountersInstrumentationEventSource.Log.IgnoreEventWrittenAsEventPayloadNotParseable(eventData.EventSource.Name);
+                }
+            }
+            catch (Exception ex)
+            {
+                EventCountersInstrumentationEventSource.Log.ErrorEventCounter(eventData.EventName, ex.ToString());
+            }
+        }
+
+        private void ExtractAndPostMetric(string eventSourceName, IDictionary<string, object> eventPayload)
+        {
+            try
+            {
+                bool calculateRate = false;
+                string actualValue = string.Empty;
+
+                string counterName = string.Empty;
+                string counterDisplayName = string.Empty;
+                InstrumentType instrumentType = InstrumentType.Counter;
+
+                foreach (KeyValuePair<string, object> payload in eventPayload)
+                {
+                    var key = payload.Key;
+                    if (key.Equals("Name", StringComparison.OrdinalIgnoreCase))
+                    {
+                        counterName = payload.Value.ToString();
+                    }
+                    else
+                    if (key.Equals("DisplayName", StringComparison.OrdinalIgnoreCase))
+                    {
+                        counterDisplayName = payload.Value.ToString();
+                    }
+                    else if (key.Equals("Mean", StringComparison.OrdinalIgnoreCase))
+                    {
+                        instrumentType = InstrumentType.Counter;
```

Comment:
this feels a bit odd.. "Mean" means the user has user EventCounter. It has nothing to do with the Otel concept of "Counter". This should be a gauge.

Reply:
You are correct. I changed it.